### ISSUE TITLE
Querying row_keys for each bucket

### ISSUE THREAD

**Issue by Yengas** [kairosdb/kairosdb#396]
This is [the line](https://github.com/kairosdb/kairosdb/blob/d890a031fa0e11c276869853982c531bc6cf4c86/src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java#L833) where KairosDB makes 1 query per bucket. And [this is](https://github.com/kairosdb/kairosdb/blob/d890a031fa0e11c276869853982c531bc6cf4c86/src/main/java/org/kairosdb/datastore/cassandra/Schema.java#L122) the query itself.

I noticed that for short period of times like 1-4 months, old schema performs better than the new one? Is this related to the additional query to `row_key_time_index` and `row_keys`?

Is it just that my computer is faster to do 1 query to the `row_key_index` and filter the results in memory, compared to doing multiple queries to Cassandra or is it a issue of caching etc?

**Comment by brianhks** [comment]
This is a good observation.  Yes it is true that it does more queries but they are done in parallel.  In a multi node C* cluster this should be faster.  How many C* nodes are you running?

**Comment by brianhks** [comment]
I just found something with the new code and I've just barely fixed it.  The key_query_time metric in the cql code is horribly inaccurate.  In the old code I had to wait until I got all the row keys back before querying for data.  The new code will start fetching data as soon as I get any row key back.  The timing code was including the time it took to send off the requests to fetch data which is wrong.  This may influence what you are seeing.
